bika/lims/browser/analysisrequest/publish.py

```diff
@@ -1345,7 +1368,10 @@ def _analysis_data(self, analysis, decimalmark=None):
                       else None,
                   'worksheet': None,
                   'specs': {},
-                  'formatted_specs': ''}
+                  'formatted_specs': '',
+                  'unit_conversions': [],
+                  'hide_primary_result': False,
```

Comment:
better inject (dict.update) that with your own function and do not rely on the key in another function

Reply:
I am extending the analysis dict with 2 new keys - this is the new dict structure. Are you saying I should initialise it in two steps?

Reply:
I would just like to avoid something like `andict['unit_conversions'].append(unit_conversion['Unit'])`, because then the reviewer and any following coder has to do the following:

- know that `andict` is a dict (easy)
- imply that the value of the key`'unit_conversion'` is a list (also easy)
- find where this key is added and check that it is always set correctly to avoid a KeyError (getting painful)
-  check where this `unit_conversion` comes from and check what is behind the `Unit` key (painful)
- find all the pieces in the dict that are relevant for the unit conversion to make a small change ...

So try to avoid explicit dictionary operation whenever possible and better make a function that returns the logical parts for a unit conversion in its own dictionary.